Add a new VertexAI error type (#8240)
firebase/firebase-js-sdk: docs-devsite/vertexai-preview.md

|  HARM\_SEVERITY\_NEGLIGIBLE | <code>&quot;HARM_SEVERITY_NEGLIGIBLE&quot;</code> |  |
|  HARM\_SEVERITY\_UNSPECIFIED | <code>&quot;HARM_SEVERITY_UNSPECIFIED&quot;</code> |  |

## VertexAIErrorCode

Standardized error codes that [VertexAIError](./vertexai-preview.vertexaierror.md#vertexaierror_class) can have.

<b>Signature:</b>

```typescript
export declare const enum VertexAIErrorCode 
```

## Enumeration Members

|  Member | Value | Description |
|  --- | --- | --- |

> |  ERROR | <code>&quot;error&quot;</code> | A generic error occured. |

Reviewer: ```suggestion
|  ERROR | <code>&quot;error&quot;</code> | A generic error occurred. |
```

Author: thanks for catching all these mistakes :p

|  FETCH\_ERROR | <code>&quot;fetch-error&quot;</code> | An error occurred while performing a fetch. |
|  INVALID\_CONTENT | <code>&quot;invalid-content&quot;</code> | An error associated with a Content object. |

> |  NO\_API\_KEY | <code>&quot;no-api-key&quot;</code> | An error occured due to a missing api key. |

Reviewer: Is this specific to the "Firebase API key"?  If so, then can we say that?

`An error occurred due to a missing Firebase API key.`

Author: Yeah, it's specific to the Firebase API key. I changed the comment to `An error occurred due to a missing Firebase API key.`.

> |  NO\_MODEL | <code>&quot;no-model&quot;</code> | An error occurred due to a missing model. |

Reviewer: ```suggestion
|  NO\_MODEL | <code>&quot;no-model&quot;</code> | An error occurred due to a model name not being specified during initialization. |
```

> |  NO\_PROJECT\_ID | <code>&quot;no-project-id&quot;</code> | An error occured due to a missing project id. |

Reviewer: ```suggestion
|  NO\_PROJECT\_ID | <code>&quot;no-project-id&quot;</code> | An error occurred due to a missing project ID. |
```

> |  PARSE\_FAILED | <code>&quot;parse-failed&quot;</code> | An error occured while parsing. |

Reviewer: ```suggestion
|  PARSE\_FAILED | <code>&quot;parse-failed&quot;</code> | An error occurred while parsing. |
```

|  REQUEST\_ERROR | <code>&quot;request-error&quot;</code> | An error occurred in a request. |

> |  RESPONSE\_ERROR | <code>&quot;response-error&quot;</code> | An error occured in a response. |

Reviewer: ```suggestion
|  RESPONSE\_ERROR | <code>&quot;response-error&quot;</code> | An error occurred in a response. |
```
